### ISSUE TITLE
Change default strategies for `paramters` to `dict_merge`

### ISSUE THREAD

**Issue by dragonraid** [Sceptre/sceptre#480]
Hello,

I propose to change default strategy for `parameters` https://github.com/cloudreach/sceptre/blob/v1.4.2/sceptre/config.py#L33
to `dict_merge` https://github.com/cloudreach/sceptre/blob/v1.4.2/sceptre/strategies.py#L16

IMO it works better with configuration cascading.
For example:
I want to deploy CF stacks to multiple AWS Regions. Those stacks are using the same CF templates and only differ in some parameters. So I want to have global configuration layer with parameters, that can be applied to all AWS Regions and then configuration file per Region with specific parameters.   

I think that, this lets you create more complex configuration directory structure and allows you to take more advantage over stack config cascading.

**Comment by ngfgrant** [comment]
Hey @dragonraid 

For version 1, we are going to keep the behaviour as `child wins`, but obviously the user can change it. The main reason we are keeping it with `child wins` is that if we changed it to say `dict merge` then it would introduce breaking changes and require a major version bump, which is already underway in master (v2) and we wanted to still include this flexibility in v1. 

I've also included the patch that enables the choice of strategy in version 2 so we can discuss it from that perspective. Another thing I will introduce as part of V2 is enabling choosing the strategies from a project-wide config file, rather than having to edit the source.

Hope that makes sense?
